### DISCUSSION TITLE
Heading Control Crazyflie 2.1

### DISCUSSION THREAD

**Discussion by rbtnav** [bitcraze/discussions#154] [Q&A]
Hello,
I am new to Crazyflie. I noticed that the magnetometer was removed on version 2.1. What is the recommended method of heading/yaw angle control on the latest version?

**Comment by knmcguire** [comment]
Hi!

Yes the magnetometer is removed but I believe that for the 2.0 it was never fused in the yaw estimate in the first place, due to the disturbances indoors. Which means that they yaw estimate comes from the kalman-filter mostly from yaw rate measurements from the gyroscope. 

This means that you can either use the [ motion commander's](https://www.bitcraze.io/documentation/repository/crazyflie-lib-python/master/api/cflib/positioning/motion_commander/)  `right()` and `left()` or the regular [commanders](https://www.bitcraze.io/documentation/repository/crazyflie-lib-python/master/api/cflib/crazyflie/commander/) `send_hover_setpoint()`

**Reply by rbtnav** [reply]
Ok so by default, Crazyflie can not hold an absolute yaw angle? Yaw would drift based on yaw gyro readings?
Thanks

**Reply by knmcguire** [reply]
the yaw estimate is pretty good, but yes, it drifts in time. The absolute yaw estimate can only be done with Mocap or lighthouse system.
